cli/test/tests/provider.test.ts

```diff
@@ -0,0 +1,191 @@
+    async () => {
+      const cwd = join("test", "tmp", "fullLifeCycle");
+      await initializeTemplate(cwd, "quickstart");
+      const providerConfig = await initProviderConfigWithPath(cwd);
+
+      assert(
+        providerConfig !== null,
+        "Provider config must already exists in a quickstart template",
+      );
+
+      const NEW_OFFER_NAME = "newOffer";
+      const [defaultOffer] = Object.values(providerConfig.offers);
+
+      assert(
+        defaultOffer !== undefined,
+        "Default offer must exist in the provider config",
+      );
+
+      providerConfig.offers = {
+        ...providerConfig.offers,
+        // add offer with remaining compute peers
+        [NEW_OFFER_NAME]: {
+          ...defaultOffer,
+          computePeers: defaultOffer.computePeers.map((_cp, i, ar) => {
+            return `nox-${numToStr(i + ar.length)}`;
+          }),
+        },
+      };
+
+      const PROVIDER_NAME = "AwesomeProvider";
+      providerConfig.providerName = PROVIDER_NAME;
+      await providerConfig.$commit();
+
+      await fluence({
+        args: ["provider", "tokens-distribute"],
+        flags: {
+          ...PRIV_KEY_1,
+          [OFFER_FLAG_NAME]: NEW_OFFER_NAME,
+          amount: "10",
+        },
+        cwd,
+      });
+
+      await fluence({ args: ["provider", "register"], flags: PRIV_KEY_1, cwd });
+      await checkProviderNameIsCorrect(cwd, PROVIDER_NAME);
+      const NEW_PROVIDER_NAME = "NewAwesomeProvider";
+      providerConfig.providerName = NEW_PROVIDER_NAME;
+      await providerConfig.$commit();
+      await fluence({ args: ["provider", "update"], flags: PRIV_KEY_1, cwd });
+      await checkProviderNameIsCorrect(cwd, NEW_PROVIDER_NAME);
+
+      await fluence({
+        args: ["provider", "offer-create"],
+        flags: {
+          ...PRIV_KEY_1,
+          [OFFER_FLAG_NAME]: NEW_OFFER_NAME,
+        },
+        cwd,
+      });
+
+      await fluence({
+        args: ["provider", "cc-create"],
+        flags: {
+          ...PRIV_KEY_1,
+          [OFFER_FLAG_NAME]: NEW_OFFER_NAME,
+        },
+        cwd,
+      });
+
+      await fluence({
+        args: ["provider", "cc-info"],
+        flags: {
+          ...PRIV_KEY_1,
+          [OFFER_FLAG_NAME]: NEW_OFFER_NAME,
+        },
+        cwd,
+      });
+
+      await fluence({
+        args: ["provider", "cc-activate"],
+        flags: {
+          ...PRIV_KEY_1,
+          [OFFER_FLAG_NAME]: NEW_OFFER_NAME,
+        },
+        cwd,
+      });
+
+      await fluence({
+        args: ["provider", "cc-info"],
+        flags: {
+          ...PRIV_KEY_1,
+          [OFFER_FLAG_NAME]: NEW_OFFER_NAME,
+        },
+        cwd,
+      });
+
+      await sleepSeconds(5);
+
+      await fluence({
+        args: ["provider", "cc-info"],
+        flags: {
+          ...PRIV_KEY_1,
+          [OFFER_FLAG_NAME]: NEW_OFFER_NAME,
+        },
+        cwd,
+      });
+
+      await sleepSeconds(CC_DURATION_MINUTES * 60);
```

Comment:
Wow, what do these sleeps mean? You actually wait so long in tests?

Reply:
It's better to mine blocks in anvil, but I think it requires more work right now